### PR TITLE
fix: properly handle mapping version of (arg)cartesian for at least dicts

### LINKED ISSUE #440: `dict` version of `ak.cartesian`
It seems the `dict` version of `ak.cartesian` is not working (as reported [here](https://github.com/CoffeaTeam/coffea/discussions/972#discussioncomment-7901390)), e.g. this `ak.cartesian({"l":l_wwz_t,"j":bjets})` would give the error pasted below. As explained by @lgray, this may be due to a mismatch of the interface of dask-awkward with awkward. 

Until the dak and ak interfaces are in line, the array version of the call (`ak.cartesian([l_wwz_t, bjets], ...)`) could be used. 

```
Traceback (most recent call last):
  File "/home/k.mohrman/miniconda3/envs/coffea23-env00/lib/python3.9/site-packages/awkward/_dispatch.py", line 51, in dispatch
    result = custom_impl(dispatch, array_likes, args, kwargs)
  File "/home/k.mohrman/miniconda3/envs/coffea23-env00/lib/python3.9/site-packages/dask_awkward/lib/core.py", line 1596, in __awkward_function__
    return fn(*args, **kwargs)
  File "/home/k.mohrman/miniconda3/envs/coffea23-env00/lib/python3.9/site-packages/dask_awkward/lib/structure.py", line 271, in cartesian
    return map_partitions(fn, *arrays, label="cartesian", output_divisions=1)
  File "/home/k.mohrman/miniconda3/envs/coffea23-env00/lib/python3.9/site-packages/dask_awkward/lib/core.py", line 1989, in map_partitions
    raise TypeError(message)
TypeError: map_partitions expects at least one Dask collection instance, you are passing non-Dask collections to dask-awkward code.
observed argument types:
- <class 'str'>- <class 'str'>

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/home/k.mohrman/coffea_dir/migrate_to_coffea2023_repo/ewkcoffea/analysis/wwz/run_wwz4l.py", line 344, in <module>
    histos_to_compute[json_name] = processor_instance.process(json_name,flist[json_name])
  File "/home/k.mohrman/coffea_dir/migrate_to_coffea2023_repo/ewkcoffea/analysis/wwz/wwz4l.py", line 572, in process
    lb_pairs = ak.cartesian({"l":l_wwz_t,"j":bjets})
  File "/home/k.mohrman/miniconda3/envs/coffea23-env00/lib/python3.9/site-packages/awkward/_dispatch.py", line 70, in dispatch
    return gen_or_result
  File "/home/k.mohrman/miniconda3/envs/coffea23-env00/lib/python3.9/site-packages/awkward/_errors.py", line 85, in __exit__
    self.handle_exception(exception_type, exception_value)
  File "/home/k.mohrman/miniconda3/envs/coffea23-env00/lib/python3.9/site-packages/awkward/_errors.py", line 95, in handle_exception
    raise self.decorate_exception(cls, exception)
TypeError: map_partitions expects at least one Dask collection instance, you are passing non-Dask collections to dask-awkward code.
observed argument types:
- <class 'str'>- <class 'str'>

This error occurred while calling

    ak.cartesian(
        {'l': dask.awkward<getitem, npartitions=1>, 'j': dask.awkward<getitem...
    )
```

### PR DESCRIPTION
**Pull request by lgray**
Fixes #440

### CODE REVIEW

**Comment by lgray** [comment]
@douglasdavis please merge and make a release when you're happy with this! Let me know if I need to add anything.

File: src/dask_awkward/lib/structure.py [hunk 1]
```diff
@@ -113,7 +112,9 @@ def argcartesian(
             behavior=behavior,
             attrs=attrs,
         )
-        return map_partitions(fn, *arrays, label="argcartesian", output_divisions=1)
+        return map_partitions(
+            fn, *arrays_packed, label="argcartesian", output_divisions=1
```
Comment (agoose77) [inline src/dask_awkward/lib/structure.py:116]:
```suggestion
            fn, *arrays_unpacked, label="argcartesian", output_divisions=1
```

File: src/dask_awkward/lib/structure.py [hunk 2]
```diff
@@ -259,7 +261,9 @@ def cartesian(
     if not highlevel:
         raise ValueError("Only highlevel=True is supported")
     if axis == 1:
+        arrays_packed, repacker = unpack_collections(arrays, traverse=True)
```
Comment (agoose77) [inline src/dask_awkward/lib/structure.py:264]:
```suggestion
        arrays_unpacked, repacker = unpack_collections(arrays, traverse=True)
```

File: src/dask_awkward/lib/structure.py [hunk 3]
```diff
@@ -268,7 +272,7 @@ def cartesian(
             behavior=behavior,
             attrs=attrs,
         )
-        return map_partitions(fn, *arrays, label="cartesian", output_divisions=1)
+        return map_partitions(fn, *arrays_packed, label="cartesian", output_divisions=1)
```
Comment (agoose77) [inline src/dask_awkward/lib/structure.py:275]:
```suggestion
        return map_partitions(fn, *arrays_unpacked, label="cartesian", output_divisions=1)
```

**Comment by douglasdavis** [comment]
Looks like Angus' comments are all taken care of?

**Pull request merged**